### PR TITLE
build: delete useless go.mod file in br/web

### PR DESCRIPTION
**Pull request by wuhuizuo**
keep only one go.mod

### What problem does this PR solve?

Issue Number: ref #37091 

Problem Summary:

### What is changed and how it works?

combine go.mod files, keep only on in repo

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: parser/go.mod [hunk 1]
```diff
@@ -1,35 +0,0 @@
-module github.com/pingcap/tidb/parser
-
```
Comment (hawkingrei) [inline parser/go.mod:2]: Please not remove this. The parser is a public library for many developer to use. it is difficult from the other components in the tidb.

Comment (wuhuizuo) [reply parser/go.mod:2]: I found failure in build pipeline 😭

File: go.mod [hunk 2]
```diff
@@ -241,7 +248,5 @@ require (
 	sigs.k8s.io/yaml v1.2.0 // indirect
 )
 
-replace github.com/pingcap/tidb/parser => ./parser
```
Comment (xhebox) [inline go.mod:244]: Please keep parser a separated package. It is intended to be a standalone component, since it may be imported by others.

**Review by xhebox — APPROVED** [review]

**Comment by xhebox** [comment]
/cc @hawkingrei

Need another review

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**